src/pseudopeople/configuration.py

```diff
@@ -0,0 +1,203 @@
+                ] = row_noise.noise_level
+        for column in form.columns:
+            if not column.is_implemented:
+                continue
+            baseline_dict[form.name][Keys.COLUMN_NOISE][column.name] = {}
+            for noise_type in column.noise_types:
+                if not noise_type.is_implemented:
+                    continue
+                baseline_dict[form.name][Keys.COLUMN_NOISE][column.name][noise_type.name] = {}
+                if noise_type.row_noise_level is not None:
+                    baseline_dict[form.name][Keys.COLUMN_NOISE][column.name][noise_type.name][
+                        Keys.ROW_NOISE_LEVEL
+                    ] = noise_type.row_noise_level
+                if noise_type.token_noise_level is not None:
+                    baseline_dict[form.name][Keys.COLUMN_NOISE][column.name][noise_type.name][
+                        Keys.TOKEN_NOISE_LEVEL
+                    ] = noise_type.token_noise_level
+                if noise_type.additional_parameters is not None:
+                    for key, value in noise_type.additional_parameters.items():
+                        baseline_dict[form.name][Keys.COLUMN_NOISE][column.name][
+                            noise_type.name
+                        ][key] = value
+        # Clean up empty layers that had no chance to `continue` out of a loop
+        if not baseline_dict[form.name][Keys.ROW_NOISE]:
+            del baseline_dict[form.name][Keys.ROW_NOISE]
+        if not baseline_dict[form.name][Keys.COLUMN_NOISE]:
+            del baseline_dict[form.name][Keys.COLUMN_NOISE]
+
+    noising_configuration.update(baseline_dict, layer="baseline")
+
+    # Update configuration with non-baseline default values
+    noising_configuration.update(DEFAULT_NOISE_VALUES, layer="default")
+
+    # Update configuration with user-supplied values
+    if user_configuration:
+        if isinstance(user_configuration, (Path, str)):
+            with open(user_configuration, "r") as f:
+                user_configuration = yaml.full_load(f)
+        user_configuration = format_user_configuration(
+            user_configuration, noising_configuration
+        )
+        noising_configuration.update(user_configuration, layer="user")
+
+    validate_noising_configuration(noising_configuration)
+
+    return noising_configuration
+
+
+def format_user_configuration(user_dict: Dict, default_config) -> Dict:
+    """Formats the user's configuration file as necessary so it can properly
+    update noising configuration to be used
+    """
+    user_dict = _format_age_miswriting_perturbations(user_dict, default_config)
+
+    return user_dict
+
+
+def _format_age_miswriting_perturbations(user_dict: Dict, default_config: ConfigTree) -> Dict:
+    # Format any age perturbation lists as a dictionary with uniform probabilites
+    for form in user_dict:
+        user_perturbations = (
+            user_dict[form]
+            .get("column_noise", {})
```

Comment:
The only change to this function (aside from moving it into a new module which is why it's all "new"/green) is to add this extra layer to the .get

Reply:
Well - had to add this "column_noise" layer a few times below as well.